bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java

```diff
@@ -1238,4 +1317,18 @@ protected void unsetSafeCaller(SafeCaller safeCaller) {
         this.safeCaller = null;
     }
 
+    @Reference
```

Comment:
As with #3617 the storage service became dynamically replaceable in all other places, then we should do so here to, shouldn't we? At least I don't see any reason to deviate from that strategy here, especially as the code is prepared for this. 

So, unless I missed anything, this should become

```java
@Reference(policy = ReferencePolicy.DYNAMIC)
```

Reply:
Implemented as suggested.